frontend/src/components/chats/components/current-chat/components/messages-list/messages-list.tsx

```diff
@@ -11,25 +12,35 @@ type Props = {
 };
 
 const MessagesList: FC<Props> = ({ messages, currentUserId }) => {
-  return (
-    <div className={styles.messagesListWrapper}>
-      {messages.map((message) => {
-        return (
-          <div key={message.id} className={styles.messageWrapper}>
-            <Message
-              messageAuthor={
-                message.sender.id === currentUserId ? 'user' : 'opponent'
-              }
-              content={message.message}
-              postTime={getFormattedDate(message.createdAt, 'HH:mm')}
-              messageAvatarUrl={
-                message.sender.userDetails.avatar?.url ?? defaultAvatar
-              }
-            />
-          </div>
-        );
-      })}
-    </div>
+  const groupedByDateMessages = groupMessagesByDate(messages);
+  const messagesToDisplay = Object.entries(groupedByDateMessages).map(
+    (group) => {
+      const [groupingDate, groupedMessages] = group;
+
+      return (
+        <div key={groupingDate}>
+          <DateSeparator postTime={groupingDate} />
+          {groupedMessages.map((message) => {
+            return (
+              <div key={message.id} className={styles.messageWrapper}>
+                <Message
+                  messageAuthor={
+                    message.sender.id === currentUserId ? 'user' : 'opponent'
+                  }
+                  content={message.message}
+                  postTime={getFormattedDate(message.createdAt, 'HH:mm')}
+                  messageAvatarUrl={
+                    message.sender.userDetails.avatar?.url ?? defaultAvatar
+                  }
+                />
+              </div>
+            );
+          })}
+        </div>
+      );
+    },
   );
+
+  return <div className={styles.messagesListWrapper}>{messagesToDisplay}</div>;
```

Comment:
```suggestion
  return <div className={styles.messagesListWrapper}>{Object.entries(groupedByDateMessages).map(}</div>;
```
avoid additional variable